Theory-on-regression.md

```diff
@@ -0,0 +1,97 @@
+# Regression Models
+
+We learned in [Introduction section](1-intro.md) that regression is to predict continuous values, and the objective of machine learning is the function or model $f(x)$, as shown in figure below.
+
+![ML-objective](image/1.1-ML-objective.png)
+
+The model $f(x)$ can also be represented as $\hat{y}(w, x)$, where $x$ is input features and $w$ is weights or parameters.
+
+Machine learning is meant to choose proper weights $w$ with [optimization algorithm](https://en.wikipedia.org/wiki/Mathematical_optimization) (learning algorithm) so that the target $\hat{y}(w, x)$ is close to real values in training examples. The "error" or "cost" between real values and predicted values is presented with [loss function](https://en.wikipedia.org/wiki/Loss_function). Then the goal of machine learning becomes minimizing the "error", i.e. loss function $J(w)$, with respect to weights $w$:
+
+$$ \color{red} \text{Goal:  } \underset{w}{\text{minimize }} J(w)$$
+
+In this section, we'll learn two regression model — the ordinary least squares for linear regression and neural network for non-linear regression — as well as related loss function and optimization algorithms.
+
+
+## Ordinary Least Squares (linear regression)
+
+Linear here means that the target value is expected to be a linear combination of the input features.
+
+### Model representation
+
+The model of linear regression is
+$$\hat{y}(w,x) = w_0 + w_1x_1 + w_2x_2 + \cdots + w_nx_n \$$
+where $\hat{y}$ is the target value, vector $X = (x_1, \cdots, x_n)$ are input features and $W = (w_0, \cdots, w_n)$ are weights. weight $w_0$ is also called intercept, and the other weights $(w_1, \cdots, w_n)$ are coefficients.
+
+If there is only one input feature, the model becomes
+$$\hat{y}(w,x) = w_0 + w_1x_1$$
```

Comment:
I would start with this, including a simple plot. This is intuitive to understand. Then make the step to multivariate linear regression.

Reply:
Good idea, updated as what you suggested.